### PR TITLE
Allow fromJSON to set keys to dirty

### LINKED ISSUE #161: unable to save Object.fromJSON() because it's not dirty
Hello!

I'm using Parse library within angular.js framework.
To bind parse object to Angular model, I'm using Parse.Object.toJSON() method.
Then, I need to update data on parse. To do this, I create new parse object using method Parse.Object.fromJSON({json}) and then call save() method.
I found that it sends to server only _updateAt field.
If I manually call "set" method on this object fields, parse starts working as expected.
I think, there is an issue with "dirty" fields.

### PR DESCRIPTION
**Pull request by dblythy**
Currently, there is no way to save fields to a Parse.Object using `fromJSON`, as `dirtyKeys` doesn't get set.

This can be confusing as at first glance, you would expect the keys to be set as they are defined. However, after the object is fetched, the keys will be null.

```
it('from json save data', async (done) => {
    const json = {
      className : 'TestObject',
      date: new Date(),
      array: [],
      object: {},
      string: ''
    }
    const obj = Parse.Object.fromJSON(json);
    expect(obj.get('date')).toBeDefined()
    expect(obj.get('date')).toBeInstanceOf(Date);
    expect(obj.get('array')).toBeDefined()
    expect(obj.get('array')).toBeInstanceOf(Array);
    expect(obj.get('object')).toBeDefined()
    expect(obj.get('object')).toBeInstanceOf(Object);
    expect(obj.get('string')).toBeDefined()
    expect(obj.get('string')).toBeInstanceOf(String);
    await obj.save();
    // all keys are defined


    await obj.fetch();
    // only created at and updated at are defined
    expect(obj.get('date')).toBeDefined()
    expect(obj.get('date')).toBeInstanceOf(Date);
    expect(obj.get('array')).toBeDefined()
    expect(obj.get('array')).toBeInstanceOf(Array);
    expect(obj.get('object')).toBeDefined()
    expect(obj.get('object')).toBeInstanceOf(Object);
    expect(obj.get('string')).toBeDefined()
    expect(obj.get('string')).toBeInstanceOf(String);
    done();
  });
```

This PR allows a 3rd parameter to `fromJSON`, where you can specify whether the keys should be set to dirty.

```js
    const json = {
      className : 'TestObject',
      date: new Date(),
      array: [],
      object: {},
      string: ''
    }
    const obj = Parse.Object.fromJSON(json, false, true);
    // now keys will be dirty
```

Properly closes #161 and #482

### CODE REVIEW

File: src/__tests__/ParseObject-test.js [hunk 1]
```diff
@@ -277,6 +277,28 @@ describe('ParseObject', () => {
     expect(o.get('date')).toBeInstanceOf(Date);
   });
 
+  it('can be dirty with fromJSON', () => {
```
Comment (dplewis) [inline src/__tests__/ParseObject-test.js:280]: Can you add an integration test?

**Review by dplewis — APPROVED** [review]
Nice! I remember this being an issue a while ago.

**Comment by ChinaeduPascal** [comment]
Why is it I cannot find the changes in the latest version `3.4.0`. The method `fromJSON` still only takes these parameters:
`fromJSON(json: any, override?: boolean): T;`.

**Comment by mtrezza** [comment]
@ChinaeduPascal This may be just a TypeScript issue. I suggest you look into the SDK code and whether the feature has been removed again. The tests in `integration/test/ParseObjectTest.js` should demo the feature. If not, please open a new issue.

**Comment by ChinaeduPascal** [comment]
@mtrezza You are right. The @types/parse definitions are not up-to-date. That's why I won't see the changes.

**Pull request merged**